documentation/source/rescap.rst

```diff
@@ -0,0 +1,120 @@
+###############################
+The cocotb ``rescap`` Testbench
+###############################
+
+.. versionadded:: 1.5
+
+This is the testbench :mod:`test_rescap` for the design ``rescap`` showing
+how cocotb can be used in an analog-mixed signal (AMS) simulation.
+
+******************
+Overview Schematic
+******************
+
+.. image:: rescap_schematic.png
+
+
+**********
+The Design
+**********
+
+The design consists of a resistor and capacitor model (both written in Verilog-AMS) connected
+in series in a SystemVerilog module, as shown here:
+
+.. code-block:: none
+
+          i_resistor             i_capacitor
+  vdd >---o--v^v^v--o------------o---||---o---< vss
+          p         n    vout    p        n
+
+
+The code of the design is:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/rescap.sv
+   :caption: rescap.sv
+   :language: systemverilog
+   :start-at: // the design-under-test
+
+
+*************
+The Testbench
+*************
+
+The testbench consists of both an :term:`HDL` part and a Python/cocotb part.
+While having :term:`HDL` as part of the testbench is not common with cocotb, it is perfectly possible.
+
+The HDL part of the Testbench
+=============================
+
+The testbench :term:`HDL` part is written in SystemVerilog and instantiates the design described above
+as ``i_rescap``.
+It also contains a probe module for analog values as instance ``i_analog_probe`` —
+imagine this being a multimeter that you quickly connect to different nodes in the design,
+measuring either voltage or current.
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/tb_rescap.sv
+   :caption: tb_rescap.sv
+   :language: systemverilog
+   :start-at: import nettypes_pkg
+
+The probe module can capture voltage and current of a node specified by ``node_to_probe``
+(a string in this module containing a hierarchical path).
+The capturing occurs whenever there is an edge on the logic signals
+``probe_voltage_toggle`` or ``probe_current_toggle``.
+The captured values can be read on real-value signals ``voltage`` and ``current`` in this module.
+
+Here is the capture code for ``voltage`` with the "user-interface" highlighted:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/analog_probe_cadence.sv
+   :caption: analog_probe_cadence.sv
+   :language: systemverilog
+   :start-at: var string node_to_probe
+   :end-at: end  // probe_voltage
+   :emphasize-lines: 1-4
+   :dedent: 2
+
+.. note:: This analog probe module is currently only implemented for the
+          Cadence Incisive and Xcelium simulators (with the AMS option available).
+          Help is appreciated to provide equivalent modules for other simulators.
```

Comment:
I would move this disclaimer up to the end of the first section, so people can set expectations before going through a page of docs.